### PR TITLE
remove userId property from UserInfo entity to avoid redudancy

### LINKED ISSUE #67: remove userId property from UserInfo entities as the entity id now holds the same value

### PR DESCRIPTION
**Pull request by poncem91**
fixes #67

### CODE REVIEW

**Comment by LeonardZ0731** [comment]
Have you tested on the functionality after making the change? I'm a little bit worrid about creating a key using id instead of keyString.

**Comment by LeonardZ0731** [comment]
Ah I think that works.

**Comment by poncem91** [comment]
> Have you tested on the functionality after making the change? I'm a little bit worrid about creating a key using id instead of keyString.

Yup! Ran all the tests and tested it out myself, but let me know if you find any bugs because of it.

File: src/test/java/com/google/neighborgood/UserInfoServletTest.java [hunk 1]
```diff
@@ -181,14 +180,14 @@ public void normalMultipleInputTest() throws IOException {
...
-
-    // The entity can't be null
-    assertNotNull(entity);
+    Key userEntityKey = KeyFactory.createKey("UserInfo", "1234567890");
+    Entity entity = null;
+    try {
+      entity = ds.get(userEntityKey);
+    } catch (EntityNotFoundException e) {
+      // The entity can't be null
+      assertNotNull(entity);
```
Comment (LeonardZ0731) [inline src/test/java/com/google/neighborgood/UserInfoServletTest.java:189]: I think the assertNotNull statement is in the wrong place?

File: src/test/java/com/google/neighborgood/UserInfoServletTest.java [hunk 2]
```diff
@@ -206,14 +205,13 @@ public void normalMultipleInputTest() throws IOException {
...
-                .setFilter(new FilterPredicate("userId", FilterOperator.EQUAL, "1234567890")));
-    entity = results.asSingleEntity();
 
-    // The entity can't be null
-    assertNotNull(entity);
+    try {
+      entity = ds.get(userEntityKey);
+    } catch (EntityNotFoundException e) {
+      // The entity can't be null
+      assertNotNull(entity);
```
Comment (LeonardZ0731) [inline src/test/java/com/google/neighborgood/UserInfoServletTest.java:213]: Same problem here. Maybe the assertNotNull statement should be placed out of the catch clause

File: src/test/java/com/google/neighborgood/UserInfoServletTest.java [hunk 3]
```diff
@@ -309,14 +307,16 @@ public void emptyInputTest() throws IOException {
     // The entity can't be null
     assertNotNull(entity);
 
+    System.out.println("\n\n\nin test. user id is: " + entity.getKey().getName());
```
Comment (LeonardZ0731) [inline src/test/java/com/google/neighborgood/UserInfoServletTest.java:310]: I think this print statement could be deleted if it is no longer needed.

Comment (poncem91) [reply src/test/java/com/google/neighborgood/UserInfoServletTest.java:310]: Oh yes, I forgot to remove this, thanks for pointing this out!

File: src/test/java/com/google/neighborgood/UserInfoServletTest.java (hunk 1 above)
Comment (poncem91) [reply src/test/java/com/google/neighborgood/UserInfoServletTest.java:189]: I'll move it outside of the catch statement :)

File: src/test/java/com/google/neighborgood/UserInfoServletTest.java (hunk 2 above)
Comment (poncem91) [reply src/test/java/com/google/neighborgood/UserInfoServletTest.java:213]: Got it, I can move it to outside of the catch statement.

File: src/main/java/com/google/neighborgood/helper/RetrieveUserInfo.java [hunk 4]
```diff
@@ -17,24 +17,21 @@
...
-import com.google.appengine.api.datastore.Query.FilterOperator;
-import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
 import com.google.appengine.api.users.UserService;
 import java.util.ArrayList;
 import java.util.List;
 
 public final class RetrieveUserInfo {
```
Comment (peap) [inline src/main/java/com/google/neighborgood/helper/RetrieveUserInfo.java:27]: Can you add some unit tests for this class?

Comment (poncem91) [reply src/main/java/com/google/neighborgood/helper/RetrieveUserInfo.java:27]: Yep! Should I go ahead and add these into this PR or create a new one for it?

Comment (LeonardZ0731) [reply src/main/java/com/google/neighborgood/helper/RetrieveUserInfo.java:27]: Ahh I have already added a test case for this class in PR #79

Comment (LeonardZ0731) [reply src/main/java/com/google/neighborgood/helper/RetrieveUserInfo.java:27]: In PR #79 there is a java file called `HelperFunctionTest.java` that tested the `RetrieveUserInfo` class and `RewardingPoints` class

Comment (poncem91) [reply src/main/java/com/google/neighborgood/helper/RetrieveUserInfo.java:27]: Ah, got it!

**Pull request merged**